### ISSUE TITLE
Cannot invoke method toInteger() on null object

### ISSUE THREAD

**Issue by pratik037** [janoodleFTW/timy-messenger#1]
I tried debugging this app after the setup from the Readme given.
But everytime I debug run the app, this error is thrown
``` Cannot invoke method toInteger() on null object ```
On checking in Android Studio, it seems the issue with app-level `build.gradle` file.
The error lies here on **Line 50**:
``` 
 def (majorNum, minorNum, buildNum) = flutterVersionName.tokenize( '.' )
  versionCode majorNum.toInteger() * 1_000_000 + minorNum.toInteger() * 1_000 + buildNum.toInteger()
```
Can you help resolving the issue?

**Comment by pratik037** [comment]
Here's my `flutter doctor -v` output:
``` Dart
[✓] Flutter (Channel beta, v1.9.1+hotfix.4, on Linux, locale en_GB.UTF-8)
    • Flutter version 1.9.1+hotfix.4 at /home/pratik037/Flutter/flutter
    • Framework revision cc949a8e8b (8 days ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/pratik037/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /home/pratik037/Android_Studio/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Android Studio (version 3.5)
    • Android Studio at /home/pratik037/Android_Studio/android-studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2019.1)
    • IntelliJ at /home/pratik037/Downloads/idea-IC-183.5912.21
    • Flutter plugin version 35.2.2
    • Dart plugin version 191.6183.88

[✓] VS Code (version 1.38.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • Redmi Note 5 Pro • 41d6ca4b • android-arm64 • Android 9 (API 28)

• No issues found!
```

**Comment by pratik037** [comment]
If I remove this line and replace it with `versionCode flutterVersionCode.toInteger()` , then **_store file_** on **line 58** gives this issue.
``` Gradle
A problem occurred evaluating project ':app'.
> path may not be null or empty string. path='null'
```

**Comment by prasadsunny1** [comment]
yes store file is giving error for me too

**Comment by nathansdev** [comment]
Same here
* What went wrong:
A problem occurred evaluating project ':app'.
> Cannot invoke method toInteger() on null object

**Comment by JMBastida** [comment]
Same here, Android app is giving that error in ios case no, I'm only able to run the app in iPhone device

**Comment by JMBastida** [comment]
> If I remove this line and replace it with `versionCode flutterVersionCode.toInteger()` , then **_store file_** on **line 58** gives this issue.
> 
> ```groovy-gradle
> A problem occurred evaluating project ':app'.
> > path may not be null or empty string. path='null'
> ```

just delete release components in gradle and error solves

**Comment by ThinkDigitalSoftware** [comment]
Many of these variables are coming from the local.properties file. It's looking for values and they're not present

**Comment by nathansdev** [comment]
yeah, issue resolved after adding our own values like thid
flutter.versionName=1.0.23
flutter.versionCode=1

**Comment by miquelbeltran** [comment]
Hey everyone, thanks for pointing the issue.

A quick temporal solution is to change `version: 1.0.23+1` to `version: 1.0.23` on the pubspec.yaml.

I have no idea why the +1 was added during the code migration to the open source project, but I guess it is because the iOS project changed the way the version is calculated.

The root of the issue is that the `versionCode` on Android is calculated from the Flutter `version` using a formula, and so it has a parsing issue if `+1` is present.

I will fix the build.gradle in the next hours. I will leave this open until then.

Thanks!

**Comment by miquelbeltran** [comment]
The issue is fixed. Keep in mind that the app still needs a complete Firebase setup to run. Let me know if you run on new issues, and again, thanks for reporting it!

**Comment by miquelbeltran** [comment]
> If I remove this line and replace it with `versionCode flutterVersionCode.toInteger()` , then **_store file_** on **line 58** gives this issue.
> 
> ```groovy-gradle
> A problem occurred evaluating project ':app'.
> > path may not be null or empty string. path='null'
> ```

BTW, this is also resolved in my PR #5 by removing the Release process we had in place and switching back to the defaults (which use debug keys)

**Comment by meddev21** [comment]
In my case I dubg in Android device using mac os, I solve this probleme by change
`android {
    compileSdkVersion localProperties.getProperty('flutter.compileSdkVersion').toInteger()
    ndkVersion flutter.ndkVersion
    .......
`
into:
`android {
    compileSdkVersion flutter.compileSdkVersion
    ndkVersion flutter.ndkVersion
    .......
`
